includes/pages/user_myshifts.php

```diff
@@ -75,17 +77,26 @@ function user_myshifts()
 
                 if ($valid) {
                     $shiftEntry->user_comment = $comment;
-                    $shiftEntry->freeloaded = $freeloaded;
-                    $shiftEntry->freeloaded_comment = $freeloaded_comment;
+                    if (
```

Comment:
its not useless

Reply:
Its is much more complicated than it has to be, the following code for example should work the same (with probably less bugs) but changes less code (as it stays with the current logical structure):

```php
if (!empty($shiftEntry)) {
            $shift = $shiftEntry->shift;
            $freeloaded_by = $shiftEntry->freeloaded_by;
            $freeloaded_comment = $shiftEntry->freeloaded_comment;

            if ($request->hasPostData('submit')) {
                $valid = true;
                if (
                    auth()->can('user_shifts_admin')
                    || $is_angeltype_supporter
                ) {
                    $newComment = strip_request_item_nl('freeloaded_comment');
                    $freeloaded_by = $request->has('freeloaded') ? ($newComment == $freeloaded_comment ? $freeloaded_by : $user->id) : null;
                    $freeloaded_comment = $newComment;
                    if ($freeloaded_by && $freeloaded_comment == '') {
                        $valid = false;
                        error(__('Please enter a freeload comment!'));
                    }
                }

                $comment = $shiftEntry->user_comment;
                $user_source = $shiftEntry->user;
                if (auth()->user()->id == $user_source->id) {
                    $comment = strip_request_item_nl('comment');
                }

                if ($valid) {
                    $shiftEntry->user_comment = $comment;
                    $shiftEntry->freeloaded_by = $freeloaded_by;
                    $shiftEntry->freeloaded_comment = $freeloaded_comment;
                    $shiftEntry->save();

                    engelsystem_log(
                        'Updated ' . User_Nick_render($user_source, true) . '\'s shift '
                        . $shift->title . ' / ' . $shift->shiftType->name
                        . ' from ' . $shift->start->format('Y-m-d H:i')
                        . ' to ' . $shift->end->format('Y-m-d H:i')
                        . ' with comment ' . $comment
                        . '. Freeloaded' . ($freeloaded_by
                            ? ' by ' . User_Nick_render($freeloaded_by, true) . ' with Comment: ' . $freeloaded_comment
                            : ': NO')
                    );
                    success(__('Shift saved.'));
                    if ($is_angeltype_supporter) {
                        throw_redirect(url('/shifts', ['action' => 'view', 'shift_id' => $shiftEntry->shift_id]));
                    }
                    throw_redirect(url('/users', ['action' => 'view', 'user_id' => $shifts_user->id]));
                }
            }

            return ShiftEntry_edit_view(
                $shifts_user,
                $shift->start->format(__('general.datetime')) . ', ' . shift_length($shift),
                $shift->location->name,
                $shift->shiftType->name,
                $shiftEntry->angelType->name,
                $shiftEntry->user_comment,
                $shiftEntry->freeloaded_by,
                $shiftEntry->freeloaded_comment,
                auth()->can('user_shifts_admin'),
                $is_angeltype_supporter
            );
        }
```

(some testing is still required while integrating it)

Reply:
$freeloaded_comment = $newComment;
reassigning freeloaded_comment before the comment was saved to newComment makes no sense for me since it is still the newComment and not the freeloaded_comment yet
same goes for freeloaded_by